### PR TITLE
Tests for various endpoints

### PR DESCRIPTION
**Pull request by Amiiiiiiiin**
Developed tests for the following postgres endpoints (#51):
* `GET /base-classes`
* `GET /characters`
* `GET /characters/place/{character_id}`
* `POST /characters`
* `PATCH /character-locations/{character_id}`

One test is missing, which has had an issue created here: https://github.com/kea-semester-1/Semester-Project/issues/69
Also see: https://github.com/kea-semester-1/Semester-Project/issues/73

### CODE REVIEW

**Comment by Moshizzl3** [comment]
Very nice! :)

**Review by Moshizzl3 — CHANGES_REQUESTED** [review]

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        # Other values
+        (-2147483647, 0),
+        (2147483646, 0),
+        (0, -2147483647),
+        (0, 2147483646),
+        (0, 0),
+        (100, 100),
+        (-100, -100),
+    ],
+)
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py:36]: I think we should store the max size of int32 in constants.py
```python
INT32_SIZE = 2**31 - 1
```

That way it will be easier to see what the test is doing, and to explain at the exam:

```suggestion
@pytest.mark.parametrize(
    "x, y",
    [
        # Int32 boundaries
        (-constants.INT32_SIZE + 1, 0),
        constants.INT32_SIZE, 0),
		...
    ],
)
```

File: rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+@pytest.mark.anyio
+async def test_get_base_classes_empty_list(client: AsyncClient) -> None:
+    """Test get all base classes with 0 base classes in the database: 200."""
+
+    response = await client.get(url)
+
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["data"] == []
+    assert len(response.json()["data"]) == 0
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py:23]: use `get_data` from test_utils to make the tests cleaner

```suggestion
@pytest.mark.anyio
async def test_get_base_classes_empty_list(client: AsyncClient) -> None:
    """Test get all base classes with 0 base classes in the database: 200."""

    response = await client.get(url)
    assert response.status_code == status.HTTP_200_OK
    
    response_data = get_data(response)
    assert response_data  == []
```

**Review by martinlaursen97 — CHANGES_REQUESTED** [review]
Nice

File: rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import pytest
+from httpx import AsyncClient
+from fastapi import status
+from rpg_api.db.postgres.factory import factories
+from typing import Any
+
+url = "/api/postgres/base-classes"
+
+
+def get_user_header(token: str) -> dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py]: Removed, as it was not used.

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+async def test_patch_character_location_valid_boundaries(
+    client: AsyncClient, x: int, y: int
+) -> None:
+    """ "Test updating character location with valid boundaries: 200."""
+
+    user = await factories.BaseUserFactory.create()
+    token = utils.create_access_token(data=dtos.TokenData(user_id=str(user.id)))
+    header = get_user_header(token)
+
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py:38]: Have converted all the tests to utilize this.

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        f"{url}/{character.id}", headers=header, json=charLocation
+    )
+
+    assert response.status_code == status.HTTP_200_OK
+    assert character.character_location.x == x
+    assert character.character_location.y == y
+
+
+@pytest.mark.anyio
+async def test_patch_character_location_missing_data(client: AsyncClient) -> None:
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py]: Changed the name of the test.

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+    response = await client.patch(
+        f"{url}/{invalidId}", headers=header, json=charLocation
+    )
+
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+
+@pytest.mark.anyio
+async def test_patch_character_location_invalid_float(client: AsyncClient) -> None:
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py]: Deleted the test as I utilised the parameterized functionality instead.

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+    await client.patch(f"{updateUrl}/{character.id}", headers=header, json=charLocation)
+
+    response = await client.get(f"{url}/{character.id}", headers=header)
+
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["data"]["name"] == "Goldshire"
+
+
+@pytest.mark.anyio
+async def test_get_character_place_named_place_outside_radius_boundary(
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:129]: Converted this and several other tests to use parameterized

File: rpg_api/tests/pytest/endpoint/characters/test_post_characters.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+@pytest.mark.anyio
+async def test_create_character_invalid_data_type(client: AsyncClient) -> None:
+    """Test character creation with invalid data types: 422."""
+
+    user = await factories.BaseUserFactory.create()
+    token = utils.create_access_token(data=dtos.TokenData(user_id=str(user.id)))
+    header = get_user_header(token)
+
+    baseClass = await factories.BaseClassFactory.create()
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_post_characters.py]: Updated this name and numerous others to use snake case instead of camel case

File: rpg_api/tests/pytest/endpoint/characters/test_post_characters.py [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    charData = {
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_post_characters.py]: Updated this name and numerous others to use snake case instead of camel case

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py (hunk 1 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py]: I did this, but I explicitly stated the values in the constants file:
INT32_MAX: int = 2147483647
INT32_MIN: int = -2147483648

File: rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py (hunk 2 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py]: Utilised it on the tests where it was relevant

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+    token = utils.create_access_token(data=dtos.TokenData(user_id=str(user.id)))
+    header = get_user_header(token)
+
+    await factories.PlaceFactory.create(
+        name="Goldshire", description="A small town", radius=10, x=25, y=-100
+    )
+    character = await factories.CharacterFactory.create(user=user)
+    # Characters are always created at 0,0 so the location is updated after creation
+    charLocation = {"x": 25, "y": -90}
+    await client.patch(f"{updateUrl}/{character.id}", headers=header, json=charLocation)
```
[the remark this answers is not in the document]
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py]: Changed this test and others where relevant to use DAO, I wasn't happy with using another endpoint in a test, but wasn't sure on how else to do it at the time.

File: rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED
+
+
+@pytest.mark.anyio
+async def test_base_classes_method_not_allowed_head(client: AsyncClient) -> None:
+    """Test that HEAD method is not allowed for the base-classes endpoint: 405."""
+
+    response = await client.head(url)
+
+    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py:105]:
```suggestion
@pytest.mark.anyio
async def test_base_classes_method_not_allowed_post(client: AsyncClient) -> None:
    """Test that POST method is not allowed for the base-classes endpoint: 405."""

    response = await client.post(url, json={"name": "NewClass"})

    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED


@pytest.mark.anyio
async def test_base_classes_method_not_allowed_put(client: AsyncClient) -> None:
    """Test that PUT method is not allowed for the base-classes endpoint: 405."""

    response = await client.put(url, json={"name": "UpdatedClass"})

    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED


@pytest.mark.anyio
async def test_base_classes_method_not_allowed_delete(client: AsyncClient) -> None:
    """Test that DELETE method is not allowed for the base-classes endpoint: 405."""

    response = await client.delete(url)

    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED


@pytest.mark.anyio
async def test_base_classes_method_not_allowed_patch(client: AsyncClient) -> None:
    """Test that PATCH method is not allowed for the base-classes endpoint: 405."""

    response = await client.patch(url, json={"name": "PatchedClass"})

    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED


@pytest.mark.anyio
async def test_base_classes_method_not_allowed_options(client: AsyncClient) -> None:
    """Test that OPTIONS method is not allowed for the base-classes endpoint: 405."""

    response = await client.options(url)

    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED


@pytest.mark.anyio
async def test_base_classes_method_not_allowed_head(client: AsyncClient) -> None:
    """Test that HEAD method is not allowed for the base-classes endpoint: 405."""

    response = await client.head(url)

    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED
```

You can probably do this in a single parameterized test.
Does this work?

```python
@pytest.mark.anyio
@pytest.mark.parametrize(
    "method, params",
    [
        ("put", {"json": {"name": "UpdatedClass"}}),
        ("delete", {}),
        ("patch", {"json": {"name": "PatchedClass"}}),
        ("options", {}),
        ("head", {})
    ],
)
async def test_base_classes_method_not_allowed(client: AsyncClient, method: str, params: dict) -> None:
    """Test that various HTTP methods are not allowed for the base-classes endpoint: 405."""

    http_method = getattr(client, method)
    response = await http_method(url, **params)

    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED
```

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+    "x, y",
+    [
+        (constants.INT32_MAX, 0),
+        (constants.INT32_MIN, 0),
+        (0, constants.INT32_MAX),
+        (0, constants.INT32_MIN),
+        (-2147483647, 0),
+        (2147483646, 0),
+        (0, -2147483647),
+        (0, 2147483646),
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py:25]: Use the INT32 constants

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+
+
+@pytest.mark.anyio
+async def test_patch_character_location_invalid_character(client: AsyncClient) -> None:
+    """Test updating character location with invalid character id: 404."""
+
+    user = await factories.BaseUserFactory.create()
+    header = test_utils.get_user_header(user.id)
+
+    invalidId = uuid.uuid4()
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py:144]:
```suggestion
    invalid_id = uuid.uuid4()
```

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+    """Test that OPTIONS method is not allowed for the endpoint: 405."""
+
+    user = await factories.BaseUserFactory.create()
+    header = test_utils.get_user_header(user.id)
+
+    character = await factories.CharacterFactory.create(user=user)
+
+    response = await client.options(f"{url}/{character.id}", headers=header)
+
+    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py:225]: Same as mentioned before

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters.py [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+    assert response_data[1]["xp"] == character2.xp
+    assert response_data[1]["money"] == character2.money
+    assert response_data[1]["base_class"].get("name") == character2.base_class.name
+    assert (
+        response_data[1]["character_location"]["x"] == character2.character_location.x
+    )
+    assert (
+        response_data[1]["character_location"]["y"] == character2.character_location.y
+    )
+    assert len(response_data) == 2
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/characters/test_get_characters.py:102]:
```suggestion
@pytest.mark.anyio
async def test_get_characters(client: AsyncClient) -> None:
    """Test get all characters: 200."""

    user = await factories.BaseUserFactory.create()
    header = test_utils.get_user_header(user.id)

    characters = [
        await factories.CharacterFactory.create(user=user),
        await factories.CharacterFactory.create(user=user)
    ]
    
    # Character not owned by current user
    await factories.CharacterFactory.create() 

    # Ensure character_location and base_class are not None for mypy
    for character in characters:
        assert character.character_location is not None
        assert character.base_class is not None

    response = await client.get(url, headers=header)
    assert response.status_code == status.HTTP_200_OK

    response_data = test_utils.get_data(response)
    
    for index, character in enumerate(characters):
        assert response_data[index]["id"] == str(character.id)
        assert response_data[index]["gender"] == character.gender
        assert response_data[index]["character_name"] == character.character_name
        assert response_data[index]["alive"] == character.alive
        assert response_data[index]["level"] == character.level
        assert response_data[index]["xp"] == character.xp
        assert response_data[index]["money"] == character.money
        assert response_data[index]["base_class"].get("name") == character.base_class.name
        assert response_data[index]["character_location"]["x"] == character.character_location.x
        assert response_data[index]["character_location"]["y"] == character.character_location.y

    assert len(response_data) == 2
```

I don't think test with an empty db / 1 characters are needed, and we can just stick with this one, since it is testing the same logic as the others anyways

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters.py [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+
+    response = await client.get(url)
+    assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+
+@pytest.mark.anyio
+async def test_characters_method_not_allowed_put(client: AsyncClient) -> None:
+    """Test that PUT method is not allowed for the base classes endpoint: 405."""
+
+    response = await client.put(url, json={"name": "Character"})
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/characters/test_get_characters.py:127]: same as mentioned before

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py [hunk 17]
```diff
@@ -0,0 +1,262 @@
...
+from httpx import AsyncClient
+from fastapi import status
+from rpg_api.db.postgres.factory import factories
+from rpg_api.tests.pytest import test_utils
+from rpg_api.utils.daos import AllDAOs
+from rpg_api.utils.dtos import CharacterLocationUpdateDTO
+import uuid
+
+url = "/api/postgres/characters/place"
+
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:11]: I think some tests in this file can be replaced with this, since it is covering the same, and is also doing boundary stuff

```python
def _url(character_id: UUID) -> str:
    return f"/api/postgres/characters/place/{character_id}"


DEFAULT_PLACE_NAME = "Goldshire"
DEFAULT_CHARACTER_X = 10


@pytest.mark.anyio
@pytest.mark.parametrize(
    "place_radius, expected_place_name",
    [
        (10.00, DEFAULT_PLACE_NAME),
        (10.01, DEFAULT_PLACE_NAME),
        (09.99, "Wilderness"),
        (09.98, "Wilderness"),
    ],
)
async def test_get_character_place_name(
    client: AsyncClient,
    place_radius: float,
    expected_place_name: str,
    daos: AllDAOs,
) -> None:
    """Test get character place name: 200."""

    user = await factories.BaseUserFactory.create()
    await factories.PlaceFactory.create(
        radius=place_radius,
        name=DEFAULT_PLACE_NAME,
        x=0,
        y=0,
    )

    character = await factories.CharacterFactory.create(
        user=user,
    )

    # TODO: remove this when character factory is fixed
    await daos.character_location.update(
        character.character_location.id,  # type: ignore
        dtos.CharacterLocationUpdateDTO(x=DEFAULT_CHARACTER_X, y=0),
    )

    header = get_user_header(user.id)

    response = await client.get(_url(character.id), headers=header)
    assert response.status_code == status.HTTP_200_OK

    response_data = get_data(response)
    assert response_data["name"] == expected_place_name
```

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py [hunk 18]
```diff
@@ -0,0 +1,262 @@
...
+
+@pytest.mark.anyio
+async def test_get_character_place_invalid_character(client: AsyncClient) -> None:
+    """Test get character in a place with invalid character id: 404."""
+
+    user = await factories.BaseUserFactory.create()
+    header = test_utils.get_user_header(user.id)
+
+    invalidId = uuid.uuid4()
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:172]:
```suggestion
    invalid_id = uuid.uuid4()
```

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py [hunk 19]
```diff
@@ -0,0 +1,262 @@
...
+
+
+@pytest.mark.anyio
+async def test_character_place_method_not_allowed_post(client: AsyncClient) -> None:
+    """Test that POST method is not allowed for the character place endpoint: 405."""
+
+    character_id = uuid.uuid4()
+
+    response = await client.post(f"{url}/{character_id}", json={"name": "NewPlace"})
+    assert response.status_code == status.HTTP_405_METHOD_NOT_ALLOWED
```
Comment (martinlaursen97) [inline rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:210]: same as mentioned before

**Review by martinlaursen97 — CHANGES_REQUESTED** [review]
Perhaps we don't need to check for invalid HTTP methods in every endpoint, but maybe a test should be written for `api/monitoring/health` that does this

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters.py (hunk 15 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters.py:102]: I think it's nice to have tests that cover for different states that the database could be in (i.e empty, single entry and multiple entries), with the only downside being a minor impact on test runtime.

@Moshizzl3 @Erol0019 Thoughts

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py (hunk 18 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:172]: Fixed

File: rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py (hunk 12 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/character_location/test_patch_character_locations.py:25]: fixed to use constant-1 and +1

File: rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py (hunk 11 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/base_classes/test_get_base_classes.py:105]: I've edited all the HTTP method tests to use this format for each disallowed method, marked the other comments regarding this as resolved.

**Comment by Amiiiiiiiin** [comment]
> Perhaps we don't need to check for invalid HTTP methods in every endpoint, but maybe a test should be written for `api/monitoring/health` that does this

I think it's fine to do for now, but we can look at it later perhaps if we're worried about duplication.

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py (hunk 17 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:11]: My thought process behind creating different tests is primarily that each test gets to have one focus for each different scenario. I believe doing it this way allows us to test each focuspoint with varying types of parameters such as negative numbers,  positive numbers, 0, small numbers, large numbers, boundaries etc.

`test_get_character_place_named_place` is focused on scenarios where a character is on the exact coordinates of a place
`test_get_character_place_named_place_radius_inside_boundary` is focused scenarios where on a character is in the radius of a place, but not on the exact coordinates
`test_get_character_place_named_place_outside_radius_boundary` is focused on testing the boundaries of a place's radius

With the change from the coordinates being int to floats, I believe we actually introduced a need for more tests (In terms of more parameters) since floats have like 6 decimals.

Comment (martinlaursen97) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:11]: For exam purposes we need to do boundary value analysis whenever possible though, and this would be the perfect test for that imo.
As the tests are written now, they are not really doing it

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters.py (hunk 15 above)
Comment (martinlaursen97) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters.py:102]: That's fair, but you should use a loop when possible to increase readability, and also create characters that are not owned by the current user, since they are not supposed to be found in this endpoint

Comment (martinlaursen97) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters.py:102]: You could even create a function at the top of the file that does the asserts, since it is a lot of duplicate code

```python
def _assert_character_data(expected_character: Character, actual_data: dict[str, Any]) -> None:
    """Asserts that the actual data from the response matches the expected character data."""
    assert actual_data["id"] == str(expected_character.id)
    assert actual_data["gender"] == expected_character.gender
    assert actual_data["character_name"] == expected_character.character_name
    assert actual_data["alive"] == expected_character.alive
    assert actual_data["level"] == expected_character.level
    assert actual_data["xp"] == expected_character.xp
    assert actual_data["money"] == expected_character.money
    assert actual_data["base_class"]["name"] == expected_character.base_class.name
    assert actual_data["character_location"]["x"] == expected_character.character_location.x
    assert actual_data["character_location"]["y"] == expected_character.character_location.y

```

then use it like this:
```python
for index, character in enumerate(characters):
	_assert_character_data(character, response_data[index])
```

Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters.py]: I've updated the test to reflect your feedback. I'll let you mark this comment as resolved in case there's anything you want to follow up on in regards to the changes made.

File: rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py (hunk 17 above)
Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:11]: All the parameters used for `test_get_character_place_named_place_radius_inside_boundary` & `test_get_character_place_named_place_outside_radius_boundary` should be the exact boundaries for the different radiuses set in different scenarios, though as if they were ints and not floats. 

We can implement your suggestion, however I am worried that we just won't get to enough scenarios as we're only updating one coordinate. That's why I made it into two different tests, where one is focused on the boundary inside the radius and one outside. Let me know what you think tomorrow 😁

Comment (Moshizzl3) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:11]: I think we should have specific tests aswell that clear tests the boundary, can't we just have both?

Comment (martinlaursen97) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:11]: I think you can keep the tests there, but also include the suggestion so we are doing boundary value analysis

Comment (Amiiiiiiiin) [reply rpg_api/tests/pytest/endpoint/characters/test_get_characters_place.py:12]: I've updated the parameters to account for the change from int to float as well as adjusted the parameters from your feedback. I'll let you mark this conversation as resolved in case you have any follow up feedback

**Review by martinlaursen97 — APPROVED** [review]

**Review by Moshizzl3 — APPROVED** [review]
